docs/modules/ROOT/pages/explanations/jsonnet.adoc

```diff
@@ -0,0 +1,91 @@
+= Jsonnet Best Practices
+
+== Merging External Config Data
+////
+* Deep merge helper ✔
+* Merge from hiera ✔
+* Expose defaults in hiera ✔
+* Dict with key=name (too keep them mergeable) ✔
+* Provide "generic" defaults and merge with context specifics (i.e. cloud/region see openshift4-nodes) ✔
+* Allow Null (to enable empty) ✔
+* For non-mergeable data (i.e. strings), provide defaults + extras ✔
+////
+If a component manages an external configuration it's considered best practice to expose this configuration in the hierarchy and to define a sensible default where possible.
+This allows to override the configuration via the hierarchy.
+
+If multiple instances of a configuration are required, expose them within a dictionary where the keys correspond to the name of an instance.
+This allows for very flexible use of a component without changing any code.
+
+Allow to set a common configuration which is used as baseline for the multiple instances.
```

Comment:
Please provide an example.

Reply:
It might be worth to provide examples both from the perspective of Jsonnet and the configuration hierarchy. Just a thought.